app/controllers/admin/reports_controller.rb

```diff
@@ -5,6 +5,14 @@ def index
 
   def show
     @report = Report.find(params[:id])
-    @post = Post.find(@report.concerns_id) if @report.concerns_model == "post"
+    begin
```

Comment:
I think it'd be better to simply check `@post.present?` instead.

Reply:
The issue is that `Post.find` doesn't return `nil` if the record is not found, it instead raises an `ActiveRecord::RecordNotFound` exception. I suppose using `Post.where.take` would prevent that issue, however.

Reply:
Perhaps using `find_by` will also do the trick, I'm not sure.

Reply:
`find_by` would probably work better yeah